service/controller/internal/nodecontroller/create.go

```diff
@@ -0,0 +1,76 @@
+package nodecontroller
+
+import (
+	"context"
+	"time"
+
+	"github.com/giantswarm/microerror"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/giantswarm/kvm-operator/service/controller/key"
+)
+
+func (c *Controller) ensureCreated(ctx context.Context, workloadNode corev1.Node) (reconcile.Result, error) {
+	c.logger.Debugf(ctx, "determining pod readiness gate based on node status")
+
+	var managementPod corev1.Pod
+	err := c.managementK8sClient.Get(ctx, key.NodePodObjectKey(c.cluster, workloadNode), &managementPod)
+	if errors.IsNotFound(err) {
+		return reconcile.Result{Requeue: false}, microerror.Mask(err)
+	} else if err != nil {
+		return reconcile.Result{
+			Requeue:      true,
+			RequeueAfter: time.Second * 10,
```

Comment:
Can we put these into an `var` somewhere? I could foresee issues where we spam too many requeue's and a single variable to change the "short" timing for all of them could be neat.

Reply:
Good idea. I turned them into constants in the `key` package https://github.com/giantswarm/kvm-operator/blob/node-controller-improved/service/controller/key/controller.go